diff --git a/kernel/linux-asic/drivers/mtd/nand/nand_ids.c b/kernel/linux-asic/drivers/mtd/nand/nand_ids.c
old mode 100644
new mode 100755
index 2e6adb1..47331a8
--- a/kernel/linux-asic/drivers/mtd/nand/nand_ids.c
+++ b/kernel/linux-asic/drivers/mtd/nand/nand_ids.c
@@ -60,6 +60,13 @@ struct nand_flash_dev nand_flash_ids[] = {
 		{ .id = {0xef, 0xaa, 0x21} },
 		  SZ_2K, SZ_128, SZ_128K, SP_OPTIONS, 3, 64,
 		  NAND_ECC_INFO(1, SZ_512), 0 },
+//by zxf
+	{"D5F2GQXX 2Gb 3.3V 8-bit",
+		{ .id = {0xc8,0x52} },
+		  SZ_2K, SZ_2G, SZ_128K, SP_OPTIONS, 2, 128,
+		  NAND_ECC_INFO(1, SZ_512), 0 },
+
+
 
 	LEGACY_ID_NAND("NAND 4MiB 5V 8-bit",   0x6B, 4, SZ_8K, SP_OPTIONS),
 	LEGACY_ID_NAND("NAND 4MiB 3,3V 8-bit", 0xE3, 4, SZ_8K, SP_OPTIONS),
@@ -191,6 +198,7 @@ struct nand_manufacturers nand_manuf_ids[] = {
 	{NAND_MFR_INTEL, "Intel"},
 	{NAND_MFR_ATO, "ATO"},
 	{NAND_MFR_WINBOND, "Winbond"},
+	{NAND_MFR_Giga, "Giga"}, //by zxd
 	{0x0, "Unknown"}
 };
 
